Sources/Datadog/Core/Persistence/Files/Directory.swift

```diff
@@ -55,11 +55,14 @@ internal struct Directory {
         // Instead of iterating over all files and removing them one by one, we create a temporary
         // empty directory and replace source directory content with (empty) temporary folder.
         // This makes the deletion atomic, and is more performant in benchmarks.
-        let temporaryDirectory = try Directory(withSubdirectoryPath: "com.datadoghq/\(UUID().uuidString)")
-        try retry(times: 3, delay: 0.001) {
-            _ = try FileManager.default.replaceItemAt(url, withItemAt: temporaryDirectory.url)
+        let tempDirectory = try Directory(withSubdirectoryPath: "com.datadoghq/\(UUID().uuidString)")
+        do {
+            try retry(times: 3, delay: 0.001) {
+                _ = try FileManager.default.replaceItemAt(url, withItemAt: tempDirectory.url)
+            }
+        } catch {
+            try? FileManager.default.removeItem(at: tempDirectory.url)
```

Comment:
I think the real issue is in this line. Why does it throw multiple times after SDK init? There must be clearly something we're missing and I'd rather vote for fixing the problem, not muting it more. It is already muted once by `try?` (we're not interested in handling this error as it doesn't seem actionable). 

Maybe this temp location doesn't exist when the runtime gets there. In such case, we can just if-check it and cleanup if necessary. WDYT?

Reply:
tl;dr: we can keep propagating the error after `removeItem(at: tempDirectory.url)`
```swift
catch {
  try? FileManager.default.removeItem(at: tempDirectory.url) // try to cleanup, we can ignore this error
  throw error // this error comes from `replaceItemAt`, this can be useful for IM
}
```
wdyt?

---
it is called once per feature (logging, tracing, rum) and 3 (or 4 if internal monitoring is enabled) times in total.
it seems like `replaceItemAt` removes `tempDirectory`:
```swift
removeItem(tempDir) // succeeds
replaceItemAt(url, tempDir)
---
replaceItemAt(url, tempDir)
removeItem(tempDir) // throws
```

my first idea was:
```swift
if fm.fileExists(at: tempDir.url) {
  fm.removeItem(at: tempDir.url)
}
```
but then i found [this in Apple docs](https://developer.apple.com/documentation/foundation/filemanager/1415645-fileexists):

> It’s far better to attempt an operation, check for errors, and handle those errors gracefully than it is to try to figure out ahead of time whether the operation will succeed.

Reply:
```swift
if FileManager.default.fileExists(atPath: temporaryDirectory.url.path) {
    try FileManager.default.removeItem(at: temporaryDirectory.url)
}
```
I don't see anything wrong with such approach. From the previous logic, we don't expect `temporaryDirectory.url` to exist (it should be removed in `mv` operation). However, if it's still there (due to unexpected / random issue in `mv`) we wan't to delete it explicitly. If that deletion causes another error, we don't have any more choices. In such case, we track it through Internal Monitoring to see how frequent it is. I don't expect we will see many reports from it.

Reply:
done